app/metal-controller-manager/api/v1alpha1/server_types.go

```diff
@@ -49,13 +61,16 @@ func (source *CredentialSource) Resolve(ctx context.Context, reader client.Clien
 		return "", nil
 	}
 
-	if source.SecretKeyRef == nil {
```

Comment:
I believe this check should stay, as one might not have the secret key ref (by mistake?)